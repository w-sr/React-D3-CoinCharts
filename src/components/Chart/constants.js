import { easeCubicOut } from "d3-ease";

// Chart
const CHART_PADDING_TOP = 20;
const TRANSITION = { duration: 500, ease: easeCubicOut };

// Cursor
const CURSOR_RADIUS_SIZE = 4;

// HorizontalAxis
const DEFAULT_TICK_COUNT = 7;

// HoverContainer
const HOVER_CONTAINER_WIDTH = 200;
const VERTICAL_OFFSET = -12;

export {
  CHART_PADDING_TOP,
  CURSOR_RADIUS_SIZE,
  DEFAULT_TICK_COUNT,
  HOVER_CONTAINER_WIDTH,
  TRANSITION,
  VERTICAL_OFFSET
};
