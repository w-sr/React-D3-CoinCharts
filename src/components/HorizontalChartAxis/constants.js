const DEFAULT_TICK_COUNT = 7;

const DURATION = {
  DAY: 'day',
  HOUR: 'hour',
  WEEK: 'week',
  MONTH: 'month',
  YEAR: 'year',
  ALL: 'all',
};

// `Object.values` polyfill for IE (since it's not supported by CRA)
const DURATION_LIST = Object.keys(DURATION).map(e => DURATION[e]);

export {
  DEFAULT_TICK_COUNT,
  DURATION,
  DURATION_LIST,
};
