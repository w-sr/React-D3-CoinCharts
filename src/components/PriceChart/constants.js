import { easeCubicOut } from "d3-ease";

// Chart
const CHART_PADDING_TOP = 20;
const TRANSITION = { duration: 500, ease: easeCubicOut };

// Cursor
const CURSOR_RADIUS_SIZE = 4;

// HorizontalAxis
const DEFAULT_TICK_COUNT = 7;

// HoverContainer
const HOVER_CONTAINER_WIDTH = 200;
const VERTICAL_OFFSET = -12;

// VerticalAxis
const ACTIVE_CURRENCY = "usd";

const DURATION = {
  DAY: "day",
  HOUR: "hour",
  WEEK: "week",
  MONTH: "month",
  YEAR: "year",
  ALL: "all"
};

// `Object.values` polyfill for IE (since it's not supported by CRA)
const DURATION_LIST = Object.keys(DURATION).map(e => DURATION[e]);

export {
  ACTIVE_CURRENCY,
  CHART_PADDING_TOP,
  CURSOR_RADIUS_SIZE,
  DEFAULT_TICK_COUNT,
  DURATION,
  DURATION_LIST,
  HOVER_CONTAINER_WIDTH,
  TRANSITION,
  VERTICAL_OFFSET
};
