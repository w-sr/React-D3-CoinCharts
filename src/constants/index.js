// Supported cryptocurrencies
const CRYPTOCURRENCY = {
  BTC: { key: 'btc', name: 'Bitcoin', fillColor: '#FFEBC5', strokeColor: '#FFB01E' },
  BCH: { key: 'bch', name: 'Bitcoin Cash', fillColor: '#e2f0d2', strokeColor: '#8DC451' },
  ETH: { key: 'eth', name: 'Ethereum', fillColor: '#F0F1F8', strokeColor: '#6E7CB9' },
  LTC: { key: 'ltc', name: 'Litecoin', fillColor: '#ECECEC', strokeColor: '#B5B5B5' },
};

// Supported Currencies
const CURRENCY = {
  cad: { key: 'cad', name: 'Canadian Dollar' },
  usd: { key: 'usd', name: 'US Dollar' },
};

// Time duration constants
const DURATION = {
  HOUR: { key: 'hour', codename: '1H', humanize: 'since an hour ago' },
  DAY: { key: 'day', codename: '1D', humanize: 'since yesterday' },
  WEEK: { key: 'week', codename: '1W', humanize: 'since last week' },
  MONTH: { key: 'month', codename: '1M', humanize: 'since last month' },
  YEAR: { key: 'year', codename: '1Y', humanize: 'since last year' },
  ALL: { key: 'all', codename: 'ALL', humanize: '' },
};

// Directory where price data (.json files) are stored
const LOCAL_JSON_DATA_DIR = './priceData';

// Polls the API endpoint every minute to update prices
const POLL_FREQUENCY = 60 * 1000;

export { CURRENCY, CRYPTOCURRENCY, DURATION, LOCAL_JSON_DATA_DIR, POLL_FREQUENCY };
