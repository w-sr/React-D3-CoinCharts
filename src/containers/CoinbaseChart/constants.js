// Supported cryptocurrencies
const CRYPTOCURRENCY = {
  BTC: { key: 'btc', name: 'Bitcoin' },
  ETH: { key: 'eth', name: 'Ethereum' },
  LTC: { key: 'ltc', name: 'Litecoin' },
};

// Supported Currencies
const CURRENCY = {
  CAD: { key: 'CAD', name: 'Canadian Dollar' },
  USD: { key: 'USD', name: 'US Dollar' },
};

// Time duration constants
const DURATION = {
  DAY: { key: 'day', codename: '1D', humanize: 'since an hour ago' },
  HOUR: { key: 'hour', codename: '1H', humanize: 'since yesterday' },
  MONTH: { key: 'month', codename: '1M', humanize: 'since last week' },
  WEEK: { key: 'week', codename: '1W', humanize: 'since last month' },
  YEAR: { key: 'year', codename: '1Y', humanize: 'since last year' },
  ALL: { key: 'all', codename: 'ALL', humanize: '' },
};

// Directory where price data (.json files) are stored
const PRICE_DATA_PATH = './priceData';

export {
  CURRENCY,
  CRYPTOCURRENCY,
  DURATION,
  PRICE_DATA_PATH,
};
