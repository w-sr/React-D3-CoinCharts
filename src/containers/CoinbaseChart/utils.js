import axios from 'axios';

import {
  PRICE_DATA_PATH,
} from './constants';

function fetchPriceData(cryptocurrency, currency, type) {
  const jsonFileName = `${cryptocurrency}_${currency}_${type}.json`;
  const jsonFilePath = `${PRICE_DATA_PATH}/${jsonFileName}`;

  return new Promise((resolve, reject) => {
    axios.get(jsonFilePath)
      .then((res) => {
        const { data } = res.data;
        resolve(data);
      })
      .catch(err => reject(err));
  });
}

export {
  fetchPriceData,
};
