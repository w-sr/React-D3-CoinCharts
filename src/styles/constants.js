export const size = {
  tiny: "2px",
  small: "4px",
  medium: "8px",
  large: "16px",
  huge: "24px",
  enormous: "32px",
};

export const fontSize = {
  tiny: "12px",
  small: "14px",
  medium: "16px",
  large: "18px",
  huge: "24px",
  mega: "48px",
};

export const fontWeight = {
  ultraLight: 100,
  thin: 200,
  light: 300,
  regular: 400,
  medium: 500,
  demiBold: 600,
  bold: 700,
  heavy: 800,
};

export const font = {
  fontFamily: `
  "Avenir Next",
  -apple-system,
  BlinkMacSystemFont,
  "Segoe UI",
  "Roboto",
  "Oxygen",
  "Ubuntu",
  "Cantarell",
  "Fira Sans",
  "Droid Sans",
  "Helvetica Neue",
  sans-serif`,
};

export const monospaceFont = {
  fontFamily: `
  "Monaco",
  monospace;`,
};

export const fontFamily = {
  regular: font.fontFamily,
  monospace: monospaceFont.fontFamily,
};

export const color = {
  /* Coincharts */
  coincharts: "#0667D0",
  coinchartsLight: "#7AA4DE",
  coinchartsGray: "#7D95B6",
  coinchartsGrayLight: "#BECADA",
  coinchartsPastel: "#BCD1EE",
  coinchartsSmoke: "#EBF1FA",

  /* Dark Grays */
  slate: "#9BA6B2",
  slateLight: "#DAE1E9",
  slateDark: "#4E5C6E",

  /* Light Grays */
  smoke: "#F4F7FA",
  smokeLight: "#F9FBFC",

  /* UI */
  neutral: "#3C90DF",
  neutralDark: "#2E7BC4",
  positive: "#61CA00",
  positiveDark: "#4BAD02",
  warning: "#F8B700",
  warningDark: "#E6A314",
  warningBlack: "#564A0E",
  negative: "#FF4949",
  negativeDark: "#E82F2F",
  moneyGreen: "#00C57F",
  moneyGreenDark: "#00AA6D",
  steel: "#1A3650",

  /* Currencies */
  bitcoin: "#FFB119",
  bitcoinLight: "#FFEBC5",
  litecoin: "#B5B5B5",
  litecoinLight: "#ECECEC",
  ethereum: "#6F7CBA",
  ethereumLight: "#F0F1F8",
  bitcoinCash: "#8DC451",
  bitcoinCashLight: "#E2F0D2",

  /* Misc. */
  white: "#FFFFFF",
  black: "#000000",
  mask: "rgba(26,54,80,0.7)",

  /* Label colors */
  labelWarning: "#FCF0BF",
  labelWarningDark: "#CCA301",
  labelPositive: "#D7F1BF",
  labelNegative: "#FFD1D1",
};

export const border = {
  border: `1px solid ${color.slateLight}`,
  noRadius: "0px",
  subtleRadius: "2px",
  borderRadius: `${size.small}`,
  radius: `${size.small}`,
  insetRadius: "3px",
};

export const boxShadow = {
  default: "0 3px 6px rgba(0, 0, 0, 0.16), 0 3px 6px rgba(0, 0, 0, 0.23)",
};

export const animation = {
  default: "all 0.25s ease",
  speed: "0.25s ease",
};

export const height = {
  dashboard: "450px",
};

export const width = {
  desktopMax: "1500px",
  desktopMin: "980px",
};

export default {
  ...color,
  ...border,
  ...fontWeight,
  ...fontSize,
  ...font,
  ...size,

  /* Dimensions */
  height,
  width,

  /* Shadow */
  boxShadow: boxShadow.default,

  /* Animation timing */
  animationDefault: animation.default,

  /* Z-index */
  zIndexDropdown: 100,
  zIndexBackdrop: -10,

  fontFamily: `"Avenir Next",
    -apple-system,
    BlinkMacSystemFont,
    "Segoe UI",
    "Roboto",
    "Oxygen",
    "Ubuntu",
    "Cantarell",
    "Fira Sans",
    "Droid Sans",
    "Helvetica Neue",
    sans-serif;`,

  monospaceFontFamily: `
    monospace;`,
};
