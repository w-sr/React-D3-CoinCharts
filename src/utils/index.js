import currencyFormatter from 'currency-formatter';

/**
 * Adds the appropriate symbol & separators to `value` based on the input `currencyCode`
 * @param {string} value
 * @param {string} currencyCode
 * @returns Formatted currency string
 */
function formatCurrency(value, currencyCode) {
  return currencyFormatter.format(value, {
    code: currencyCode.toUpperCase(),
  });
}

// eslint-disable-next-line import/prefer-default-export
export { formatCurrency };
