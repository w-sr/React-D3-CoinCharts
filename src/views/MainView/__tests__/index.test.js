import React from "react";

import { renderWithReduxStore } from "../../../utils/tests";
import { width } from "../../../styles/constants";
import { UnconnectedMainView } from "../";

const DESKTOP_TEST_WIDTH = width.desktopMin + 100;
const MOBILE_TEST_WIDTH = width.desktopMin - 100;
const TEST_PROPS = {
  requestPriceData: () => {},
};

describe("<MainView/>", () => {
  it("[MOBILE] renders without crashing", () => {
    global.innerWidth = MOBILE_TEST_WIDTH;
    const { container } = renderWithReduxStore(<UnconnectedMainView {...TEST_PROPS} />);
    expect(container.firstChild).not.toBeNull();
    expect(container.firstChild).toMatchSnapshot();
  });

  it("[DESKTOP] renders without crashing", () => {
    global.innerWidth = DESKTOP_TEST_WIDTH;
    const { container } = renderWithReduxStore(<UnconnectedMainView {...TEST_PROPS} />);
    expect(container.firstChild).not.toBeNull();
    expect(container.firstChild).toMatchSnapshot();
  });

  describe("Price data polling", () => {
    // Use fake timers to tests `setInterval()` behavior
    jest.useFakeTimers();

    it("starts polling for price data when rendered", () => {
      const props = {
        requestPriceData: jest.fn(),
      };

      renderWithReduxStore(<UnconnectedMainView {...props} />);
      expect(props.requestPriceData).toHaveBeenCalledTimes(0);

      jest.runOnlyPendingTimers();
      expect(props.requestPriceData).toHaveBeenCalledTimes(1);

      jest.runOnlyPendingTimers();
      expect(props.requestPriceData).toHaveBeenCalledTimes(2);

      jest.runOnlyPendingTimers();
      expect(props.requestPriceData).toHaveBeenCalledTimes(3);
    });

    it("stops polling for price data when component is unmounted", () => {
      const props = {
        requestPriceData: jest.fn(),
      };

      const { unmount } = renderWithReduxStore(<UnconnectedMainView {...props} />);
      expect(props.requestPriceData).toHaveBeenCalledTimes(0);

      jest.runOnlyPendingTimers();
      expect(props.requestPriceData).toHaveBeenCalledTimes(1);
      unmount();

      jest.runOnlyPendingTimers();
      expect(props.requestPriceData).toHaveBeenCalledTimes(1);

      jest.runOnlyPendingTimers();
      expect(props.requestPriceData).toHaveBeenCalledTimes(1);
    });
  });
});
