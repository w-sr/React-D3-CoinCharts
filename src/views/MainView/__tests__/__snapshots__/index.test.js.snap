// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MainView/> [DESKTOP] renders without crashing 1`] = `
<div
  class="sc-jWBwVP jIipSk sc-bdVaJa kiEYim"
>
  <svg
    class="sc-ifAKCX fWKbFX"
    viewBox="0 0 250 250"
    aria-hidden="true"
  >
    <defs>
      <mask
        id="hole"
        fill="#000"
      >
        <path
          d="M0,0 L115,115 L130,115 L142,142 L250,250 L250,0 Z"
          mask="url(#hole)"
          fill="#fff"
        />
        <path
          class="sc-bxivhb bUZGRr"
          d="M128.3,109.0 C113.8,99.7 119.0,89.6 119.0,89.6 C122.0,82.7 120.5,78.6 120.5,78.6 C119.2,72.0 123.4,76.3 123.4,76.3 C127.3,80.9 125.5,87.3 125.5,87.3 C122.9,97.6 130.6,101.9 134.4,103.2"
        />
        <path
          d="M115.0,115.0 C114.9,115.1 118.7,116.5 119.8,115.4 L133.7,101.6 C136.9,99.2 139.9,98.4 142.2,98.6 C133.8,88.0 127.5,74.4 143.8,58.0 C148.5,53.4 154.0,51.2 159.7,51.0 C160.3,49.4 163.2,43.6 171.4,40.1 C171.4,40.1 176.1,42.5 178.8,56.2 C183.1,58.6 187.2,61.8 190.9,65.4 C194.5,69.0 197.7,73.2 200.1,77.6 C213.8,80.2 216.3,84.9 216.3,84.9 C212.7,93.1 206.9,96.0 205.4,96.6 C205.1,102.4 203.0,107.8 198.3,112.5 C181.9,128.9 168.3,122.5 157.7,114.1 C157.9,116.9 156.7,120.9 152.7,124.9 L141.0,136.5 C139.8,137.7 141.6,141.9 141.8,141.8 Z"
        />
      </mask>
    </defs>
    <a
      class="sc-EHOje cKGmQE"
      href="https://github.com/test-user/test-project"
      target="_blank"
      aria-label="View source on Github"
    >
      <path
        d="M0,0 L115,115 L130,115 L142,142 L250,250 L250,0 Z"
        mask="url(#hole)"
      />
    </a>
  </svg>
  <div
    class="sc-kkGfuU iOEVQQ sc-bdVaJa hHhwCB"
  >
    <div
      class="sc-cvbbAY bHDGhx sc-bdVaJa bJonzb"
    >
      <div
        class="sc-hSdWYo igXijG sc-bdVaJa kDHipt"
      >
        Coincharts
      </div>
      <div
        class="sc-bZQynM foTJIu sc-bdVaJa cAcgKX"
        role="tabpanel"
      >
        <div
          class="sc-gzVnrw jRDvTs sc-bdVaJa dyxdRy"
          color="#7D95B6"
          aria-labelledby="1H"
          aria-selected="false"
          role="tab"
          tabindex="-1"
        >
          <span>
            1H
          </span>
        </div>
        <div
          class="sc-gzVnrw kOMeBY sc-bdVaJa dyxdRy"
          color="#0667D0"
          aria-labelledby="1D"
          aria-selected="true"
          role="tab"
          tabindex="-1"
        >
          <span>
            1D
          </span>
        </div>
        <div
          class="sc-gzVnrw jRDvTs sc-bdVaJa dyxdRy"
          color="#7D95B6"
          aria-labelledby="1W"
          aria-selected="false"
          role="tab"
          tabindex="-1"
        >
          <span>
            1W
          </span>
        </div>
        <div
          class="sc-gzVnrw jRDvTs sc-bdVaJa dyxdRy"
          color="#7D95B6"
          aria-labelledby="1M"
          aria-selected="false"
          role="tab"
          tabindex="-1"
        >
          <span>
            1M
          </span>
        </div>
        <div
          class="sc-gzVnrw jRDvTs sc-bdVaJa dyxdRy"
          color="#7D95B6"
          aria-labelledby="1Y"
          aria-selected="false"
          role="tab"
          tabindex="-1"
        >
          <span>
            1Y
          </span>
        </div>
        <div
          class="sc-gzVnrw jRDvTs sc-bdVaJa dyxdRy"
          color="#7D95B6"
          aria-labelledby="ALL"
          aria-selected="false"
          role="tab"
          tabindex="-1"
        >
          <span>
            ALL
          </span>
        </div>
      </div>
    </div>
    <div
      class="sc-eHgmQL AEESZ"
    >
      <div
        class="sc-eNQAEJ hiCQLY sc-bdVaJa hpOpwR"
      >
        <div>
          <div
            class="sc-kGXeez iIWYYw"
          >
            <span
              color="#4E5C6E"
              class="sc-kpOJdX kdHsge"
            >
              $
            </span>
            <span>
              0
            </span>
            <span
              color="#4E5C6E"
              class="sc-kpOJdX kdHsge"
            >
              .00
            </span>
          </div>
          <div
            color="#4E5C6E"
            transform="rotate(180deg)"
            class="sc-kEYyzF lkEVfq"
          >
            <svg
              height="12"
              width="12"
              viewBox="8135 12432 15 15"
              version="1.1"
              xmlns="http://www.w3.org/2000/svg"
            >
              <title>
                Arrow
              </title>
              <desc>
                Arrow pointing downwards
              </desc>
              <path
                fill-rule="evenodd"
                d="M8149.72 12439.36l-7.36-7.36-7.36 7.36 1 1 5.64-5.64v11.82h1.4v-11.87l5.68 5.69z"
              />
            </svg>
            <span
              data-testid="price-details"
            >
              $0.00 (0.00%)
            </span>
          </div>
        </div>
        <select
          class="sc-hMqMXs iOLeJY"
        >
          <option
            value="BTC"
          >
            BTC
          </option>
          <option
            value="BCH"
          >
            BCH
          </option>
          <option
            value="ETH"
          >
            ETH
          </option>
          <option
            value="LTC"
          >
            LTC
          </option>
        </select>
      </div>
      <div>
        <div
          class="sc-kgoBCf jfrcOa sc-bdVaJa cAcgKX"
        >
          <div
            class="sc-cSHVUG kQaYsT sc-bdVaJa ivjahH"
            data-testid="VerticalAxis"
          />
          <div
            class="sc-chPdSV cklyAr"
            data-testid="HoverRegion"
          >
            <svg
              class="sc-gqjmRU hyAmY"
              data-testid="Graph"
            >
              <g />
            </svg>
          </div>
        </div>
        <div
          class="sc-VigVT cYFHMG sc-bdVaJa bJonzb"
          data-testid="HorizontalAxis"
        />
      </div>
    </div>
  </div>
  <div
    class="sc-bwzfXH jIYfKf"
  >
    <span>
      Powered by 
    </span>
    <a
      href="https://developers.coinbase.com/api/v2"
      target="_blank"
      rel="noopener noreferrer"
    >
      <img
        class="sc-htpNat pSHAW"
        src="coinbase-logo.svg"
        alt="coinbase-logo"
      />
    </a>
  </div>
</div>
`;

exports[`<MainView/> [MOBILE] renders without crashing 1`] = `
<div
  class="sc-jWBwVP jIipSk sc-bdVaJa kiEYim"
>
  <svg
    class="sc-ifAKCX fWKbFX"
    viewBox="0 0 250 250"
    aria-hidden="true"
  >
    <defs>
      <mask
        id="hole"
        fill="#000"
      >
        <path
          d="M0,0 L115,115 L130,115 L142,142 L250,250 L250,0 Z"
          mask="url(#hole)"
          fill="#fff"
        />
        <path
          class="sc-bxivhb bUZGRr"
          d="M128.3,109.0 C113.8,99.7 119.0,89.6 119.0,89.6 C122.0,82.7 120.5,78.6 120.5,78.6 C119.2,72.0 123.4,76.3 123.4,76.3 C127.3,80.9 125.5,87.3 125.5,87.3 C122.9,97.6 130.6,101.9 134.4,103.2"
        />
        <path
          d="M115.0,115.0 C114.9,115.1 118.7,116.5 119.8,115.4 L133.7,101.6 C136.9,99.2 139.9,98.4 142.2,98.6 C133.8,88.0 127.5,74.4 143.8,58.0 C148.5,53.4 154.0,51.2 159.7,51.0 C160.3,49.4 163.2,43.6 171.4,40.1 C171.4,40.1 176.1,42.5 178.8,56.2 C183.1,58.6 187.2,61.8 190.9,65.4 C194.5,69.0 197.7,73.2 200.1,77.6 C213.8,80.2 216.3,84.9 216.3,84.9 C212.7,93.1 206.9,96.0 205.4,96.6 C205.1,102.4 203.0,107.8 198.3,112.5 C181.9,128.9 168.3,122.5 157.7,114.1 C157.9,116.9 156.7,120.9 152.7,124.9 L141.0,136.5 C139.8,137.7 141.6,141.9 141.8,141.8 Z"
        />
      </mask>
    </defs>
    <a
      class="sc-EHOje cKGmQE"
      href="https://github.com/test-user/test-project"
      target="_blank"
      aria-label="View source on Github"
    >
      <path
        d="M0,0 L115,115 L130,115 L142,142 L250,250 L250,0 Z"
        mask="url(#hole)"
      />
    </a>
  </svg>
  <div
    class="sc-iAyFgw cPwqga sc-kkGfuU iOEVQQ sc-bdVaJa hHhwCB"
  >
    <div
      class="sc-cvbbAY bHDGhx sc-bdVaJa bJonzb"
    >
      <div
        class="sc-bZQynM foTJIu sc-bdVaJa cAcgKX"
        role="tabpanel"
      >
        <div
          class="sc-gzVnrw kOMeBY sc-bdVaJa dyxdRy"
          color="#0667D0"
          aria-labelledby="Bitcoin-$0.00"
          aria-selected="true"
          role="tab"
          tabindex="-1"
        >
          <span
            class="sc-htoDjs kegmoF"
          >
            <span>
              Bitcoin
            </span>
            <span>
              $0.00
            </span>
          </span>
        </div>
        <div
          class="sc-gzVnrw jRDvTs sc-bdVaJa dyxdRy"
          color="#7D95B6"
          aria-labelledby="Bitcoin Cash-$0.00"
          aria-selected="false"
          role="tab"
          tabindex="-1"
        >
          <span
            class="sc-htoDjs kegmoF"
          >
            <span>
              Bitcoin Cash
            </span>
            <span>
              $0.00
            </span>
          </span>
        </div>
        <div
          class="sc-gzVnrw jRDvTs sc-bdVaJa dyxdRy"
          color="#7D95B6"
          aria-labelledby="Ethereum-$0.00"
          aria-selected="false"
          role="tab"
          tabindex="-1"
        >
          <span
            class="sc-htoDjs kegmoF"
          >
            <span>
              Ethereum
            </span>
            <span>
              $0.00
            </span>
          </span>
        </div>
        <div
          class="sc-gzVnrw jRDvTs sc-bdVaJa dyxdRy"
          color="#7D95B6"
          aria-labelledby="Litecoin-$0.00"
          aria-selected="false"
          role="tab"
          tabindex="-1"
        >
          <span
            class="sc-htoDjs kegmoF"
          >
            <span>
              Litecoin
            </span>
            <span>
              $0.00
            </span>
          </span>
        </div>
      </div>
      <div
        class="sc-bZQynM foTJIu sc-bdVaJa cAcgKX"
        role="tabpanel"
      >
        <div
          class="sc-gzVnrw jRDvTs sc-bdVaJa dyxdRy"
          color="#7D95B6"
          aria-labelledby="1H"
          aria-selected="false"
          role="tab"
          tabindex="-1"
        >
          <span>
            1H
          </span>
        </div>
        <div
          class="sc-gzVnrw kOMeBY sc-bdVaJa dyxdRy"
          color="#0667D0"
          aria-labelledby="1D"
          aria-selected="true"
          role="tab"
          tabindex="-1"
        >
          <span>
            1D
          </span>
        </div>
        <div
          class="sc-gzVnrw jRDvTs sc-bdVaJa dyxdRy"
          color="#7D95B6"
          aria-labelledby="1W"
          aria-selected="false"
          role="tab"
          tabindex="-1"
        >
          <span>
            1W
          </span>
        </div>
        <div
          class="sc-gzVnrw jRDvTs sc-bdVaJa dyxdRy"
          color="#7D95B6"
          aria-labelledby="1M"
          aria-selected="false"
          role="tab"
          tabindex="-1"
        >
          <span>
            1M
          </span>
        </div>
        <div
          class="sc-gzVnrw jRDvTs sc-bdVaJa dyxdRy"
          color="#7D95B6"
          aria-labelledby="1Y"
          aria-selected="false"
          role="tab"
          tabindex="-1"
        >
          <span>
            1Y
          </span>
        </div>
        <div
          class="sc-gzVnrw jRDvTs sc-bdVaJa dyxdRy"
          color="#7D95B6"
          aria-labelledby="ALL"
          aria-selected="false"
          role="tab"
          tabindex="-1"
        >
          <span>
            ALL
          </span>
        </div>
      </div>
    </div>
    <div
      class="sc-eHgmQL AEESZ"
    >
      <div
        class="sc-dxgOiQ iqvuHD sc-bdVaJa gObvDU"
      >
        <div
          class="sc-ckVGcZ jNkpwx sc-bdVaJa ZqsgD"
        >
          <div
            class="sc-kGXeez iIWYYw"
          >
            <span
              color="#4E5C6E"
              class="sc-kpOJdX kdHsge"
            >
              $
            </span>
            <span>
              0
            </span>
            <span
              color="#4E5C6E"
              class="sc-kpOJdX kdHsge"
            >
              .00
            </span>
          </div>
          <div
            class="sc-jKJlTe dkUzDh"
          >
            Bitcoin
            price
          </div>
        </div>
        <div
          class="sc-ckVGcZ jNkpwx sc-bdVaJa ZqsgD"
        >
          <div
            class="sc-kGXeez iIWYYw"
          >
            <span
              color="#4E5C6E"
              class="sc-kpOJdX kdHsge"
            >
              $
            </span>
            <span>
              0
            </span>
            <span
              color="#4E5C6E"
              class="sc-kpOJdX kdHsge"
            >
              .00
            </span>
          </div>
          <div
            class="sc-jKJlTe dkUzDh"
          >
            since yesterday (USD)
          </div>
        </div>
        <div
          class="sc-ckVGcZ jNkpwx sc-bdVaJa ZqsgD"
        >
          <div
            class="sc-kGXeez iIWYYw"
          >
            <span>
              0.00
            </span>
            <span
              color="#4E5C6E"
              class="sc-kpOJdX kdHsge"
            >
              %
            </span>
          </div>
          <div
            class="sc-jKJlTe dkUzDh"
          >
            since yesterday
            (%)
          </div>
        </div>
      </div>
      <div>
        <div
          class="sc-kgoBCf jfrcOa sc-bdVaJa cAcgKX"
        >
          <div
            class="sc-cSHVUG kQaYsT sc-bdVaJa ivjahH"
            data-testid="VerticalAxis"
          />
          <div
            class="sc-chPdSV cklyAr"
            data-testid="HoverRegion"
          >
            <svg
              class="sc-gqjmRU hyAmY"
              data-testid="Graph"
            >
              <g />
            </svg>
            <svg
              style="opacity: 0;"
              class="sc-dnqmqq dVvouu"
              data-testid="Cursor"
            >
              <line
                class="sc-gZMcBi eeKCXY"
                x1="-1"
                x2="-1"
                y1="0"
                y2="0"
              />
              <circle
                class="sc-iwsKbI fAKhyv"
                cx="-1"
                cy="-1"
              />
            </svg>
            <div
              class="sc-fjdhpX guKSnK sc-bdVaJa epQZNM"
              style="top: -12px; left: -101px; opacity: 0;"
              data-testid="HoverContainer"
            >
              <div
                background="#7D95B6"
                border="none"
                color="#FFFFFF"
                class="sc-jzJRlG etfxMZ"
              />
            </div>
            <div
              class="sc-fjdhpX guKSnK sc-bdVaJa epQZNM"
              style="bottom: -12px; left: -101px; opacity: 0;"
              data-testid="HoverContainer"
            >
              <div
                background="#FFFFFF"
                border="1px solid #DAE1E9"
                color="#7D95B6"
                class="sc-jzJRlG iJCuCR"
              />
            </div>
          </div>
          <div
            class="sc-cSHVUG hbWElG sc-bdVaJa kAcxML"
            data-testid="VerticalAxis"
          />
        </div>
        <div
          class="sc-VigVT qQVXw sc-bdVaJa bJonzb"
          data-testid="HorizontalAxis"
        />
      </div>
    </div>
  </div>
  <div
    class="sc-bwzfXH jIYfKf"
  >
    <span>
      Powered by 
    </span>
    <a
      href="https://developers.coinbase.com/api/v2"
      target="_blank"
      rel="noopener noreferrer"
    >
      <img
        class="sc-htpNat pSHAW"
        src="coinbase-logo.svg"
        alt="coinbase-logo"
      />
    </a>
  </div>
</div>
`;
